03-rnaseq/pathway-analysis_rnaseq_01_ora.Rmd

```diff
@@ -0,0 +1,522 @@
+module_19_genes <- module_annot_df %>%
+  dplyr::filter(module == "ME19") %>%
+  dplyr::pull("gene_symbol")
+```
+
+Because one `gene_symbol` may map to multiple Ensembl IDs, we need to make sure we have no repeated gene symbols in this list.
+
+```{r}
+# Reduce to only unique gene symbols
+genes_of_interest <- unique(as.character(module_19_genes))
+
+# Let's print out some of these genes
+head(genes_of_interest)
+```
+
+## Determine our background set gene list
+
+Sometimes folks consider genes from the entire genome to comprise the background, but for this RNA-seq example, we will consider all detectable genes as our background set.
+The dataset that these genes are from already had unreliably detected, [low count genes removed it](https://alexslemonade.github.io/refinebio-examples/04-advanced-topics/network-analysis_rnaseq_01_wgcna.html#prepare-data-for-deseq2). 
+Because of this, we can obtain our detected genes list from our data frame, `module_annot_df` (which we have not done any further filtering on in this notebook).
+
+```{r}
+background_set <- unique(as.character(module_annot_df$gene_symbol))
+```
+
+**IDENTICAL TO MICROARRAY EXAMPLE** (except plot names changed and results summaries altered)
+
+## Run ORA using the `enricher()` function
+
+Now that we have our background set, our genes of interest, and our pathway information, we're ready to run ORA using the `enricher()` function.
+
+```{r}
+kegg_ora_results <- enricher(
+  gene = genes_of_interest, # A vector of your genes of interest
+  pvalueCutoff = 0.1, # Can choose a FDR cutoff
+  pAdjustMethod = "BH", # What method for multiple testing correction should we use
+  universe = background_set, # A vector containing your background set genes
+  # The pathway information should be a data frame with a term name or
+  # identifier and the gene identifiers
+  TERM2GENE = dplyr::select(
+    hs_kegg_df,
+    gs_name,
+    gene_symbol
+  )
+)
+```
+
+*Note: using `enrichKEGG()` is a shortcut for doing ORA using KEGG, but the approach we covered here can be used with any gene sets you'd like!*
+
+The information we're most likely interested in is in the `results` slot.
+Let's convert this into a data frame that we can write to file.
+
+```{r}
+kegg_result_df <- data.frame(kegg_ora_results@result)
+```
+
+Let's print out a sneak peek of it here and take a look at how many sets do we have that fit our cutoff of `0.1` FDR?
+
+```{r}
+kegg_result_df %>%
+  dplyr::filter(p.adjust < 0.1)
+```
+
+Looks like there are four KEGG sets returned as significant at FDR of `0.1`.
+
+## Visualizing results
+
+We can use a dot plot to visualize our significant enrichment results.
+The `enrichplot::dotplot()` function will only plot gene sets that are significant according to the multiple testing corrected p values (in the `p.adjust` column) and the `pvalueCutoff` you provided in the [`enricher()` step](#run-ora-using-the-enricher-function).
+
+```{r}
+enrich_plot <- enrichplot::dotplot(kegg_ora_results)
```

Comment:
Any idea why the following message is showing up at this step? 

`## wrong orderBy parameter; set to default orderBy = "x"`

Reply:
I'm not getting this message. Do you have the most recent docker image?

Reply:
Oh I see it. Hmm...

Reply:
This is an error that is known in this version of clusterProfiler: https://github.com/YuLab-SMU/enrichplot/issues/22
It looks like if we bumped up to a more recent version we may not have this problem. The users may not encounter this issue depending on what version they install.

Reply:
Because it doesn't show up in the render and users may not encounter it and it doesn't seem to affect the plot, I'm inclined to let this one go. But it was good to look into. :+1: